profiling/src/collections/string_storage.rs

```diff
@@ -32,13 +65,24 @@ impl ManagedStringStorage {
             id_to_data: Default::default(),
             str_to_id: Default::default(),
             current_gen: 0,
+            next_cached_profile_id: InternalCachedProfileId { id: 0 },
         };
         // Ensure empty string gets id 0 and always has usage > 0 so it's always retained
         // Safety: On an empty managed string table intern should never fail.
         storage.intern_new("").expect("Initialization to succeed");
         storage
     }
 
+    pub fn next_cached_profile_id(&mut self) -> anyhow::Result<CachedProfileId> {
+        let next_id = self.next_cached_profile_id.id;
+        self.next_cached_profile_id = InternalCachedProfileId {
+            id: next_id
+                .checked_add(1)
+                .ok_or_else(|| anyhow::anyhow!("Ran out of cached_profile_ids!"))?,
```

Comment:
`.context("Ran out of cached_profile_ids!")`

Reply:
Fixed in 4ee0c0be1061da71fd48fa433d668ddc3a21a076